src/webgpu/api/validation/capability_checks/features/query_types.spec.ts

```diff
@@ -10,38 +10,35 @@ export const g = makeTestGroup(ValidationTest);
 g.test('createQuerySet')
   .desc(
     `
-Tests that creating query set shouldn't be valid without the required feature enabled.
-- createQuerySet
-  - type {occlusion, timestamp}
-  - x= {pipeline statistics, timestamp} query {enable, disable}
-
-TODO: This test should expect *synchronous* exceptions, not validation errors, per
-<https://github.com/gpuweb/gpuweb/blob/main/design/ErrorConventions.md>.
-As of this writing, the spec needs to be fixed as well.
+  Tests that creating a query set throws a type error exception if the features don't contain
+  'timestamp-query'.
+    - createQuerySet
+      - type {occlusion, timestamp}
+      - x= {pipeline statistics, timestamp} query {enable, disable}
   `
   )
   .params(u =>
     u
       .combine('type', ['occlusion', 'timestamp'] as const)
-      .combine('timestampQueryEnable', [false, true])
+      .combine('featureContainsTimestampQuery', [false, true])
   )
   .beforeAllSubcases(t => {
-    const { timestampQueryEnable } = t.params;
+    const { featureContainsTimestampQuery } = t.params;
 
     const requiredFeatures: GPUFeatureName[] = [];
-    if (timestampQueryEnable) {
+    if (featureContainsTimestampQuery) {
       requiredFeatures.push('timestamp-query');
     }
 
     t.selectDeviceOrSkipTestCase({ requiredFeatures });
   })
   .fn(async t => {
-    const { type, timestampQueryEnable } = t.params;
+    const { type, featureContainsTimestampQuery } = t.params;
 
     const count = 1;
-    const shouldError = type === 'timestamp' && !timestampQueryEnable;
+    const shouldException = type === 'timestamp' && !featureContainsTimestampQuery;
 
-    t.expectValidationError(() => {
+    t.shouldThrow(shouldException ? 'TypeError' : false, () => {
```

Comment:
When type="timestamp" and featureContainsTimestampQuery=false, the test fails because a validation error is generated instead of a type error exception. I guess that Chromium doesn't generate an exception yet. Anyway, I update the code to expect an exception since the specification says it should throw a type error exception in such a case. But, I wonder if changing to expect an exception makes sense here now. wdyt?

Reply:
This test looks correct. We should always update the CTS, even when Chromium is outdated, as we are creating a conformance target for Chromium and all other browsers to aim for.

(We always want CTS to match the latest agreed-upon standard, but unfortunately sometimes the spec has errors or is outdated relative to the standard. So when you find a difference between the spec and Chromium, it's good to double check which one is wrong, by checking for open issues on crbug.com, crbug.com/dawn, or gpuweb/gpuweb.)

Reply:
ok, I will keep working on the task like this. Thanks!

Reply:
In support of what Kai said, as we get closer to releasing the API the gaps between the spec and the implementations will become less obvious. Having visible failures in the CTS helps guide our implementation efforts.

Thank you!